services/greasyfork/greasyfork-downloads.service.js

```diff
@@ -0,0 +1,58 @@
+import { renderDownloadsBadge } from '../downloads.js'
+import BaseGreasyForkService from './greasyfork-base.js'
+
+class BaseGreasyForkInstalls extends BaseGreasyForkService {
+  static category = 'downloads'
+
+  static defaultBadgeData = { label: 'installs' }
+}
+
+class GreasyForkDailyInstalls extends BaseGreasyForkInstalls {
+  static route = { base: 'greasyfork/dd', pattern: ':scriptId' }
+
+  static examples = [
+    {
+      title: 'Greasy Fork',
+      namedParams: { scriptId: '407466' },
+      staticPreview: this.render({ downloads: 17 }),
+    },
+  ]
+
+  static _cacheLength = 21600
+
+  static render({ downloads }) {
+    return renderDownloadsBadge({ downloads, interval: 'day' })
+  }
+
+  async handle({ scriptId }) {
+    const data = await this.fetch({ scriptId })
+    return this.constructor.render({
+      downloads: data.daily_installs,
+    })
+  }
+}
+
+class GreasyForkTotalInstalls extends BaseGreasyForkInstalls {
+  static route = { base: 'greasyfork/dt', pattern: ':scriptId' }
+
+  static examples = [
+    {
+      title: 'Greasy Fork',
+      namedParams: { scriptId: '407466' },
+      staticPreview: this.render({ downloads: 3420 }),
+    },
+  ]
+
+  static _cacheLength = 21600
+
+  static render({ downloads }) {
+    return renderDownloadsBadge({ downloads })
+  }
+
+  async handle({ scriptId }) {
+    const data = await this.fetch({ scriptId })
+    return this.constructor.render({ downloads: data.total_installs })
+  }
+}
+
+export { GreasyForkDailyInstalls, GreasyForkTotalInstalls }
```

Comment:
This can and should be combined into a single class, both because it creates opportunities for a richer user experience on Shields.io similar to other services with multiple variants, and to avoid code duplication.

The only variable parts are the field in the response and the `interval` value to pass to the render helper which can be handled via inline logic. Having a single class with a variant-type route parameter will handle exposing both options to end users and be more consistent with what we do for other classes. Some examples for reference

https://github.com/badges/shields/blob/2d7d8c5b784cbcb7f2728a818d7c36a73dc71030/services/gem/gem-downloads.service.js#L28

https://github.com/badges/shields/blob/2d7d8c5b784cbcb7f2728a818d7c36a73dc71030/services/crates/crates-downloads.service.js#L9